src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -124,6 +126,655 @@ const submitEncryptModeForm: ControllerHandler<
     return res.status(statusCode).json({ message: errorMessage })
   }
 
+  // Retrieve public key.
+  const publicKey = formResult.value.publicKey
+
+  if (!publicKey) {
+    logger.warn({
+      message: 'Form does not have a public key',
+      meta: logMeta,
+    })
+    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Form does not have a public key',
+    })
+  }
+
+  setFormTags(formResult.value)
+
+  const checkFormIsEncryptModeResult = checkFormIsEncryptMode(formResult.value)
+  if (checkFormIsEncryptModeResult.isErr()) {
+    logger.error({
+      message:
+        'Trying to submit non-encrypt mode submission on encrypt-form submission endpoint',
+      meta: logMeta,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      checkFormIsEncryptModeResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+  const form = checkFormIsEncryptModeResult.value
+
+  // Check that form is public
+  const formPublicResult = FormService.isFormPublic(form)
+  if (formPublicResult.isErr()) {
+    logger.warn({
+      message: 'Attempt to submit non-public form',
+      meta: logMeta,
+      error: formPublicResult.error,
+    })
+    const { statusCode, errorMessage } = mapRouteError(formPublicResult.error)
+    if (statusCode === StatusCodes.GONE) {
+      return res.sendStatus(statusCode)
+    } else {
```

Comment:
Qn:
- Does the frontend convert 410 status to appropriate error message?
- Might want to consider standardising and return error message directly from backend
(same for submitStorageModeForm)

Reply:
Nope, the frontend does not. I'm assuming this has to do with the [message on formsg-clowning](https://opengovproducts.slack.com/archives/C019EQSBLTD/p1691974800148029). Okay! Will just return the error message from backend ("This form is no longer active").

Reply:
Haha yes admittedly I paid attention to this because of the slack convo 😅 
but I did find it unusual that GONE was a special case where we did not return an error message explicitly
thanks for fixing!